spantest/inmemory.go

```diff
@@ -0,0 +1,79 @@
+package spantest
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"path/filepath"
+	"strconv"
+	"testing"
+
+	"cloud.google.com/go/spanner"
+	"cloud.google.com/go/spanner/spannertest"
+	"cloud.google.com/go/spanner/spansql"
+	"google.golang.org/api/option"
+	"google.golang.org/grpc"
+	"gotest.tools/v3/assert"
+)
+
+// InMemoryFixture is a test fixture running the Spanner emulator.
+type InMemoryFixture struct {
+	Ctx context.Context
+}
+
+// NewInMemoryFixture creates a test fixture for the in-memory Spanner emulator.
+func NewInMemoryFixture(t *testing.T) *InMemoryFixture {
+	t.Helper()
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+	if deadline, ok := t.Deadline(); ok {
+		ctx, cancel = context.WithDeadline(ctx, deadline)
+		t.Cleanup(cancel)
+	}
```

Comment:
is it necessary to "re-apply" the deadline here?

Reply:
Hmm, maybe not - I'll remove it!